pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,56 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+	regionCfg := &aws.Config{Region: aws.String(dsInfo.Region)}
+	cfgs := []*aws.Config{
+		regionCfg,
+	}
+	// Choose authentication scheme based on the type chosen for the data source
+	// Basically, we support the following methods:
+	// Shared credentials: Providing access key pair sourced from user's AWS credentials file
+	// Static credentials: Providing access key pair directly
+	// SDK: Leave it to SDK to decide
+	switch dsInfo.AuthType {
+	case "credentials":
```

Comment:
Seems like these strings are being used in tests and so forth so would suggest adding constants/an enum.